python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -891,17 +891,29 @@ def _with_updated_values(
         )
         return out
 
-    def _resolve_and_update_env_vars(self) -> "ConfigurableResourceFactory[TResValue]":
-        """Processes the config dictionary to resolve any EnvVar values. This is called at runtime
-        when the resource is initialized, so the user is only shown the error if they attempt to
-        kick off a run relying on this resource.
+    def _resolve_and_update_configuration_values(
+        self, context: InitResourceContext
+    ) -> "ConfigurableResourceFactory[TResValue]":
+        """Processes the config dictionary attached to the resource to resolve any EnvVar values. As well
+        as overrides any values passed in via run_config (attached to the InitResourceContext).
+
+        This is called at runtime when the resource is initialized, so the user is only shown
+        the errors involving environment variable resolution if they attempt to kick off a
+        run relying on this resource.
 
         Returns a new instance of the resource.
         """
+        # This is the configuration passed via run_config (e.g. launchpad or Python API)
+        incoming_resource_config = process_config(self._schema.config_type, context.resource_config)
+
```

Comment:
This poses a real problem in the case of enums. This gets post-processed with actually converts the enum to be the value, rather than the name, of the enum value. When that in turn is passed back to the ConfigurableResourceFactory constructor it does another roundtrip through the config validator, which fails the validation.